src/util/loading.cpp

```diff
@@ -16,62 +16,110 @@ void
 load_manifest() {
     auto manifest_folder = files::get_manifest_path();
 
-    world::entity::ObjectHandler& object_handler =
-        world::entity::ObjectHandler::instance();
-
     for (const auto& directory_entry :
          std::filesystem::directory_iterator(manifest_folder)) {
-        Json::Value manifest;
+        auto manifest_opt =
+            files::read_json_from_file<manifest::manifest_t>(directory_entry.path());
 
-        // path to manifest file
-        auto manifest_file = directory_entry.path();
+        if (!manifest_opt.has_value()) {
+            continue;
+        }
 
-        auto contents = files::open_data_file(manifest_file);
-        if (contents.has_value()) [[likely]] {
-            // might fail
-            // if the json has some problems this will throw
+        manifest::manifest_t& manifest = manifest_opt.value();
+
+        int num_biomes =
+            manifest.biomes.has_value() ? manifest.biomes.value().size() : 0;
+        int num_entities =
+            manifest.entities.has_value() ? manifest.entities.value().size() : 0;
+
+        LOG_DEBUG(
+            logging::file_io_logger, "Loading manifest {} with {} biomes, {} entities.",
+            manifest.name, num_biomes, num_entities
+        );
+        if (manifest.entities.has_value()) {
+            // iterate through objects in manifest and queue them to be loaded
+            for (const manifest::descriptor_t& entity_data :
+                 manifest.entities.value()) {
+                // Will check if path exists
+                GlobalContext& context = GlobalContext::instance();
+                auto future = context.submit([entity_data]() {
+                    world::entity::ObjectHandler& object_handler =
+                        world::entity::ObjectHandler::instance();
+                    object_handler.read_object(entity_data);
+                });
+            }
+        }
+        // if (manifest.biomes.has_value()) -> do something else
+    }
+}
 
-            Json::CharReaderBuilder builder;
-            builder["collectComments"] = false;
+// I know this entire things is copied. I know it's bad.
+// It is like this because I don't know if I can
+//  1 use opengl without a screen (GitHub)
+//  2 conveniently get errors back
+int
+load_manifest_test() {
```

Comment:
update this with comments from above

Reply:
Sorry this wasn't clear in the comments. The difference between `load_manifest` and `load_manifest_test` is that load_manifest_test does not use opengl, and each "task" is run in series and if any fail the function returns.

Reply:
No i mean update this function with the comments i made on `load_manifest` lol